### DISCUSSION TITLE
[AzureDevops] Hosted agent usage

### DISCUSSION THREAD

**Discussion by shessane** [actions/virtual-environments#8265] [General questions] [answered]
Hi,

Can you please share with us where the runner-images are deployed ? Can we use those images to configure a vmss agent pool ?

Many thanks

**Comment by shamil-mubarakshin** [comment]
Hey @shessane. Images from this repository are used as [GitHub-hosted runners](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners) and [Microsoft-hosted agents](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops#use-a-microsoft-hosted-agent).

You can use these images to configure VMSS. One of the approaches would be with [this guide](https://github.com/actions/runner-images/blob/main/docs/create-image-and-azure-resources.md), where you can use repository scripts to build managed image in Azure and use it for VMSS creation.

**Reply by shessane** [reply]
Hi @shamil-mubarakshin, thank you for the quick reply.

My exact question is about using the same image from your stores. Is there any chance that you publish those images to a public gallery that we can use without creating our images ? If not, is there any chance that this can be planned in future ?
The idea is really to have the exact same images used by GitHub Actions or Microsoft hosted agents, but from a pool that has access to a private VNet. This will align our release pipelines (Github pipelines, Azure devops with Microsoft hosted pipeline and Azure devops with self hosted pipelines)

Maintaining a release process for the images based on the source code with the same quality (Beta/GA/Latest) will be time consuming and redundant with your work.

Thank you very much

**Reply by shamil-mubarakshin** [reply, marked as answer]
As of now, there are no plans on making internally built images publicly accessible
